UsersApp/Networking/NetworkMonitor.swift

```diff
@@ -0,0 +1,34 @@
+import UIKit
+import Foundation
+import Network
+
+
+final class NetworkMonitor {
+    static let shared = NetworkMonitor()
+    
+    private let monitor: NWPathMonitor
+    private (set) var isConnected = false
+    private (set) var isExpensive = false
+    private (set) var currentConnectionType: NWInterface.InterfaceType?
+    private let networkQueue = DispatchQueue(label: "NetworkConnectivityMonitor")
+    
+    private init() {
+        monitor = NWPathMonitor()
+    }
+    
+    func startMonitoring() {
+        monitor.pathUpdateHandler = { [weak self] path in
+            DispatchQueue.main.async {
+                self?.isConnected = path.status != .unsatisfied
+                self?.isExpensive = path.isExpensive
```

Comment:
I don't think that we need .isExpensive property or type of network, but it's good to have

Reply:
yes, initially I didn't use it in the sceneDelegate, but I was thinking that it may cover an edge case I didn't think about.